gm4_liquid_minecarts/data/gm4_liquid_minecarts/functions/init_tank.mcfunction

```diff
@@ -1,22 +1,29 @@
 #@s = liquid minecart
 #run from try_to_unload
 
-data modify block ~ ~ ~ CustomName set from entity @s CustomName
+#summon display stand
 summon armor_stand ~ ~-.45 ~ {CustomName:"\"gm4_liquid_tank_display\"",Tags:["gm4_no_edit","gm4_liquid_tank_display","gm4_lm_needs_texture"],NoGravity:1,Marker:1,Invisible:1,Invulnerable:1,Small:1,DisabledSlots:2039583,Fire:20000}
-data modify entity @e[type=armor_stand,tag=gm4_lm_needs_texture,limit=1,sort=nearest] ArmorItems[3] set from entity @e[type=armor_stand,tag=gm4_liquid_minecart_display,limit=1,sort=nearest] HandItems[0]
+data modify entity @e[type=armor_stand,tag=gm4_lm_needs_texture,limit=1,sort=nearest] ArmorItems[3] set from entity @s Passengers[0].ArmorItems[0].tag.gm4_liquid_minecarts.tank.texture
+data modify entity @e[type=armor_stand,tag=gm4_liquid_tank,limit=1,sort=nearest] ArmorItems[3].tag.gm4_liquid_tanks.liquid_tag set from entity @s Passengers[0].ArmorItems[0].tag.gm4_liquid_minecarts.liquid_tag
 tag @e[type=armor_stand] remove gm4_lm_needs_texture
+
+#setup tank data
+data modify block ~ ~ ~ CustomName set from entity @s Passengers[0].ArmorItems[0].tag.gm4_liquid_minecarts.tank.CustomName
 scoreboard players operation @s gm4_lm_data = @s gm4_lt_max
 scoreboard players operation @s gm4_lm_data *= #5 gm4_lm_data
 scoreboard players operation @s gm4_lm_data /= #2 gm4_lm_data
 scoreboard players operation @e[type=armor_stand,tag=gm4_liquid_tank,limit=1,sort=nearest,distance=..0.5] gm4_lt_max = @s gm4_lm_data
+
+#apply liquid tag
+data modify entity @e[type=armor_stand,tag=gm4_liquid_tank,limit=1,sort=nearest,distance=..0.5] Tags append from entity @s Passengers[0].ArmorItems[0].tag.gm4_liquid_minecarts.liquid_tag
+tag @e[type=armor_stand,tag=gm4_liquid_tank,limit=1,sort=nearest,distance=..0.5] remove gm4_lt_empty
```

Comment:
You are doing more modifications and on this newly created entity. The same problem as described above could happen. Since you need both entities, it's not an option to run a function `as` this new armor stand. I suggest you keep the `gm4_lm_needs_texture` tag around until the end of this function and maybe rename it.

Reply:
Same as my last comment. Do you think it might be worth it to give this tank a tag early in the function and only refer to it by that, or would that not decrease the number of selectors

Reply:
If you use a distance check, a tag would not be required to optimize. So using `@e[type=armor_stand,tag=gm4_liquid_tank,limit=1,sort=nearest,distance=..0.5]` is fine.